subprojects/launcher/src/main/java/org/gradle/launcher/daemon/configuration/DaemonParameters.java

```diff
@@ -177,7 +177,7 @@ public DaemonParameters setBaseDir(File baseDir) {
     }
 
     public boolean getDebug() {
-        return jvmOptions.getDebug();
+        return userDefinedImmutableJvmArgs ? true : jvmOptions.getDebug();
```

Comment:
why did this change?

Reply:
A test was failing that verified this specific behavior. I can have a deeper look again but that was pretty much the behavior before.